scheduler/drilling-wits-depth-median/src/app.py

```diff
@@ -0,0 +1,70 @@
+from typing import Dict
+
+from corva import Api, Cache, Logger, ScheduledDepthEvent, scheduled
+
+from src import api as api_lib
+from src import models, service
+from src.configuration import SETTINGS
+
+
+@scheduled
+def app(event: ScheduledDepthEvent, api: Api, cache: Cache):
+    records = api_lib.get_records(
+        api=api,
+        asset_id=event.asset_id,
+        log_identifier=event.log_identifier,
+        top_depth=event.top_depth,
+        bottom_depth=event.bottom_depth,
+    )
+
+    if not records:
+        Logger.info('Fetched zero records. Returning early.')
+        return
+
+    milestone_to_records = service.group_records_by_milestone(
+        records=records, interval=event.interval
+    )
+
+    out_dataset = SETTINGS.OUT_DATASET.format(interval=event.interval)
+
+    existing_summaries = api_lib.get_summaries(
+        api=api,
+        asset_id=event.asset_id,
+        log_identifier=event.log_identifier,
+        measured_depths=list(milestone_to_records),
```

Comment:
It's a little surprising that we can convert to `list` and it'll automatically know which field is the measured depth field, right? Is this correct?

Reply:
`milestone_to_records` is a dict, where keys are milestones and values are list of record associated with that milestone.
We look up if there are existing documents for current milestones. To get all current milestones (all keys) we cast dict to list.
list({"key1": 1, "key2": 2}) == ["key1", "key2"]

Reply:
Can you leave this as a comment in the code?